api/v1alpha08/sonataflowplatform_services_types.go

```diff
@@ -0,0 +1,93 @@
+	// +optional
+	Persistence *PersistenceOptions `json:"persistence,omitempty"`
+}
+
+// ServicesPodTemplateSpec describes the desired custom Kubernetes PodTemplate definition for the deployed flow.
+//
+// The ContainerSpec describes the container where the actual service is running. It will override any default definitions.
+// For example, to override the image one can use `.spec.services.dataIndex.container.image = my/image:tag`.
+type ServicesPodTemplateSpec struct {
+	// Container is the Kubernetes container where the service should run.
+	// One can change this attribute in order to override the defaults provided by the operator.
+	// +optional
+	Container *ContainerSpec `json:"container,omitempty"`
+	// +optional
+	PodSpec `json:",inline"`
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+	// Determines whether "prod" profile workflows should be configured to use this service
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+}
+
+// PersistenceOptions configure the services to persist to a datasource of choice
+// +kubebuilder:validation:MaxProperties=1
+type PersistenceOptions struct {
+	// Connect configured services to a postgresql database.
+	// +optional
+	PostgreSql *PersistencePostgreSql `json:"postgresql,omitempty"`
+}
+
+// PersistencePostgreSql configure postgresql connection for service(s).
+// +kubebuilder:validation:MinProperties=2
+// +kubebuilder:validation:MaxProperties=2
+type PersistencePostgreSql struct {
+	// Secret reference to the database user credentials
+	SecretRef PostgreSqlSecretOptions `json:"secretRef"`
+	// Service reference to postgresql datasource. Mutually exclusive to jdbcUrl.
+	// +optional
+	ServiceRef PostgreSqlServiceOptions `json:"serviceRef,omitempty"`
+	// PostgreSql JDBC URL. Mutually exclusive to serviceRef.
+	// e.g. "jdbc:postgresql://host:port/database?currentSchema=data-index-service"
+	// +optional
+	JdbcUrl string `json:"jdbcUrl,omitempty"`
+}
+
+// PostgreSqlSecretOptions use credential secret for postgresql connection.
+type PostgreSqlSecretOptions struct {
+	// Name of the postgresql credentials secret.
+	Name string `json:"name"`
+	// Defaults to POSTGRESQL_USER
+	// +optional
+	UserKey string `json:"userKey,omitempty"`
+	// Defaults to POSTGRESQL_PASSWORD
+	// +optional
+	PasswordKey string `json:"passwordKey,omitempty"`
+}
+
+// PostgreSqlServiceOptions use k8s service to configure postgresql jdbc url.
+type PostgreSqlServiceOptions struct {
+	// Name of the postgresql k8s service.
+	Name string `json:"name"`
+	// Namespace of the postgresql k8s service. Defaults to the SonataFlowPlatform's local namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+	// Port to use when connecting to the postgresql k8s service. Defaults to 5432.
+	// +optional
+	Port *int `json:"port,omitempty"`
+	// Name of postgresql database to be used. Defaults to "sonataflow"
+	// +optional
+	DatabaseName string `json:"databaseName,omitempty"`
```

Comment:
Maye we can add and optional attribute DatabaseSchema string, in cases users wants to use a particular one.
By doing this, we can build the same url as the JdbcUrl case: jdbc:postgresql://host:port/database?**currentSchema=data-index-service**"

Reply:
I like this. Actually, if we are going to share the same service this is a **must**. Can we do it, @tchughesiv?

Reply:
If we expect to share DB between jobs service and data index, I think it's a good idea to let them use different schemas, among other thing to now have conflicts en the flyway table with the table versions for each service.

Reply:
Exactly. Very nice catch!

Reply:
@ricardozanini if that's the case, maybe it's better not to share persistence configs between service types. maybe, as @wmedvede stated, we should move persistence config to the data index object now?

Reply:
@tchughesiv yes, indeed. We can either:
- Keep the persistence config here and add a `persistence.schema` in the data index object or
- Move the persistence config to data index, as stated.

Scenarios:
- Users wanna have two different instances, so all good they keep them separated in each structure
- Users wanna have one instance, two schemas, then have to duplicate info in both services

I think I'd rather move to data index, and later have one in jobs services. Users can duplicate info, it's less error prune. 

wdyt guys?

Reply:
agreed

Reply:
@ricardozanini sorry but I don't understood this last proposal, what are you suggesting to move to the data-index

The "Persistence" or adding filed "persistence.schema"  ?

Reply:
this thread became a combination of two discussions. i've added the optional DatabaseSchema attribute, and i've also moved the broader persistence config to DI. This will allow for custom persistence and/or schema config(s) for each service type going forward (e.g. job service)